### PR TITLE
Windows Server 2012 Install task

### PR DESCRIPTION
**Pull request by tannoa2**
The tasks set up the network config (static/DHCP), and the user name and password.
The task is missing the VLAN configuration

### CODE REVIEW

**Comment by tannoa2** [comment]
test this please

**Comment by tannoa2** [comment]
test this please

File: lib/task-data/tasks/install-windows-2012.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+'use strict';
+
+module.exports = {
+    friendlyName: 'Install Windows',
+    injectableName: 'Task.Os.Install.Win',
+    implementsTask: 'Task.Base.Os.Install',
+    options: {
+            osType: 'windows', //readonly option, should avoid change it
+
+            profile: 'winpe.ipxe',
```
Comment (anhou) [inline lib/task-data/tasks/install-windows-2012.js]: Is 'winpe' naming still appropriate here? maybe 'window' or 'windowsserver' or some other nams more accurate?

File: lib/task-data/tasks/install-windows-2012.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            domain: 'rackhd',
+            password: "onrack1",
+            rootSshKey: null,
+            username: "onrack",
+            "networkDevices": [],
+            dnsServers: [],
+            installDisk: null,
+            smbUser: 'onrack',
+            smbPassword: "onrack",
+            server: "172.31.128.1",
```
Comment (zyoung51) [inline lib/task-data/tasks/install-windows-2012.js]: {{api.server}} instead of hardcoded value

File: lib/task-data/tasks/install-windows-2012.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    implementsTask: 'Task.Base.Os.Install',
+    options: {
+            osType: 'windows', //readonly option, should avoid change it
+
+            profile: 'windows.ipxe',
+            completionUri: 'winpe-kickstart.ps1',
+            version: null,
+            hostname: 'localhost',
+            domain: 'rackhd',
+            password: "onrack1",
```
Comment (zyoung51) [inline lib/task-data/tasks/install-windows-2012.js]: The other installation tasks default to: "RackHDRocks!"

**Comment by tannoa2** [comment]
thanks @zyoung51

**Comment by keedya** [comment]
:+1:

**Comment by anhou** [comment]
:+1:

**Pull request merged**